cli/src/cli.rs

```diff
@@ -942,86 +945,118 @@ pub fn process_command(config: &CliConfig) -> ProcessResult {
-                            exit(1);
-                        }),
-                    ),
-                    ConnectionCache::Quic(cache) => Arc::new(
-                        TpuClient::new_with_connection_cache(
-                            rpc_client.clone(),
-                            &config.websocket_url,
-                            TpuClientConfig::default(),
-                            cache,
-                        )
-                        .unwrap_or_else(|err| {
-                            eprintln!("Could not create TpuClient {err:?}");
-                            exit(1);
-                        }),
-                    ),
-                }
+            let connection_cache = if config.use_tpu_client {
+                Some({
+                    #[cfg(feature = "dev-context-only-utils")]
+                    let cache = ConnectionCache::new_quic_for_tests(
+                        "connection_cache_cli_ping_quic",
+                        DEFAULT_TPU_CONNECTION_POOL_SIZE,
+                    );
+                    #[cfg(not(feature = "dev-context-only-utils"))]
+                    let cache = ConnectionCache::new_quic(
+                        "connection_cache_cli_ping_quic",
+                        DEFAULT_TPU_CONNECTION_POOL_SIZE,
+                    );
+                    cache
+                })
             } else {
-                rpc_client.clone() as Arc<dyn TpsClient + 'static>
+                None
             };
-            process_ping(
-                &client_dyn,
-                config,
-                interval,
-                count,
-                timeout,
-                blockhash,
-                *print_timestamp,
-                *compute_unit_price,
-                &rpc_client,
-            )
+
+            match connection_cache {
+                Some(ConnectionCache::Quic(cache)) => {
+                    let tpu_client = TpuClient::new_with_connection_cache(
+                        rpc_client.clone(),
+                        &config.websocket_url,
+                        TpuClientConfig::default(),
+                        cache,
+                    )
+                    .await
+                    .unwrap_or_else(|err| {
+                        eprintln!("Could not create TpuClient {err:?}");
+                        std::process::exit(1);
+                    });
+
+                    process_ping(
+                        Some(&tpu_client),
+                        config,
+                        interval,
+                        count,
+                        timeout,
+                        blockhash,
+                        *print_timestamp,
+                        *compute_unit_price,
+                        &rpc_client,
+                    )
+                    .await
+                }
+                Some(ConnectionCache::Udp(cache)) => {
+                    let tpu_client = TpuClient::new_with_connection_cache(
```

Comment:
Since this code path is not reachable, we could simplify `process_ping` to specify that it only works with quic TPU clients, and avoid all the generic params on it.

But to be honest, let's just land this first.